### PR TITLE
`Paywalls`: separate `PurchasesHybridCommonUI`

### PR DESCRIPTION
**Pull request by NachoSoto**
Depends on https://github.com/RevenueCat/purchases-hybrid-common/pull/672

See also https://github.com/RevenueCat/purchases-flutter/pull/943

### CODE REVIEW

File: react-native-purchases-ui/RNPaywalls.podspec [hunk 1]
```diff
@@ -17,6 +17,7 @@ Pod::Spec.new do |spec|
   spec.pod_target_xcconfig = { 'DEFINES_MODULE' => 'YES' }
 
   spec.dependency   "React-Core"
-  spec.dependency   "PurchasesHybridCommon", '8.12.0'
+  spec.dependency   "PurchasesHybridCommon", '9.0.0'
```
Comment (tonidero) [inline react-native-purchases-ui/RNPaywalls.podspec:20]: Same as in the flutter one, I wonder if we could remove this dependency, even if you would still need the base SDK to do most things.

**Review by tonidero — APPROVED** [review]
Looks good! Same as in the flutter PR, just a comment.

**Review by vegaro — APPROVED** [review]

**Comment by vegaro** [comment]
Going to merge this so I can rebase my branch

File: react-native-purchases-ui/RNPaywalls.podspec (hunk 1 above)
Comment (NachoSoto) [reply react-native-purchases-ui/RNPaywalls.podspec:20]: Done: #864

**Pull request merged**